### DISCUSSION TITLE
Need help setting up the toolchain for Mac OS Sonoma

### DISCUSSION THREAD

**Discussion by aarnink** [picocomputer/community#97] [Q&A]
Hi Community,

Over the last days I build a PicoComputer as well and everything works! 

Now for the next step I want to learn more about coding for the RP6502. I watched the tutorial for Linux but I am not able to get the toolchain running on Mac OS. 

I am not very experienced with VSCode. 

Here is where I am currently:
1.  In Github cloned the template to my own repository which is also on my local drive using GitHUB desktop to sync. 

2.  installed brew components:
- brew install cc65
- brew install cmake  
- brew install python3
- brew install pip (not needed is part of python it states)
- python3 -m pip install pyserial   

4. install VSCode

5. in VSCode installed the ca65  macro assembler and CMake (which VSCode offered to do itself)

then in VSCode when I navigate to to src in VSCode and open main.c the editor shows me an error 
_"cannot open source file "rp6502.h"C/C++(1696)"_

Can anyone help me setup the toolchain for Mac please?

Regards, Richard

**Comment by vruumllc** [comment]
The error means that the version of cc65 installed by brew doesn't yet support the rp6502.  It is best to use the repository here:
https://github.com/picocomputer/cc65 and then build it yourself. Sorry, I'm not a Mac person that can help you with the build and install, but on Linux, all it takes is 'make', then 'sudo make install'.

Maybe this will help too?
https://github.com/cc65/wiki/wiki
 
If that fails you might want to try installing the llvm-mos compiler instead.  It is a much better compiler than cc65, and it is pretty trivial to change the cc65 projects into llvm-mos.  Start with this sample project and then move the src files and modify the main CMakeLists.txt:
https://github.com/picocomputer/vscode-llvm-mos

If the llvm-mos compiler you download doesn't support the rp6502 yet either, you can download the SDK from here:
https://github.com/picocomputer/llvm-mos-sdk

Good Luck. Sorry I don't have a Mac to test things out.
tonyvr

**Reply by gmixich** [reply]
Thank you. I had same problem in MacOS as described by [aarnink].  Used cc65 https://github.com/picocomputer/cc65  repo and it worked.

**Reply by vruumllc** [reply]
Glad to hear you are up and running! Check out my repo for lots of goodies to try:
https://github.com/vruumllc

**Reply by aarnink** [reply]
Ok I am a step closer, thank you very much it works for me as well. As I am not too familiair with coding on mac and vscode in general it took me quite some time. 

**I will try to explain what I did to make it work:**
1.  do not use brew, install the command line tools from xcode   (brew did not allow me to install using pip  pyserial)
2. /Library/Developer/CommandLineTools/usr/bin/python3 -m pip install pyserial
3. install github desktop (made it easier for me to deal with github repos)
4. install vscode
5. use the repo [gmixich] suggested above (I cloned this to my local folder and then run make from the terminal)
6. use the template to create a copy of the[vscode-cc65](https://github.com/picocomputer/vscode-cc65). template as described  and clone this to your local folder (I used github desktop_ 
7. in github desktop open VSCode on the repo you just cloned on your local folder
8. in VScode locate settings.json (in folder .vscode) and change SystemIncludePath, add the folder which contains the CA65 which you created earlier in step 5)

When I run the Build command, I get a file named hello.rp6502 which is created in the build folder . I can copy this to the USB stick and run it from the commandline of the rp6502.  I cannot get it running from the terminal window automatically (as shown in EP9 of the C programming setup clip on youtube).

Hello World is working, so next is the Star Swarms from @[vruumllc]
Unfortunately that does not work. VScode  is not able to find the build instructions , If I navigate in the menu to Terminal/Run Build Task it says **"no task to build found. Confgure Build Task.."**. What do I need to do to overcome this issue?

**Reply by aarnink** [reply]
@vruumllc  the toolchain on mac works now, I downloaded star_swarms to check if I can build this game. I noticed that both the cc65 and llvm-mos version of the code are in the repo in different subfolders. In my case this does not work, the build task is not recognized. I got it working after I moved the cc65 folder contents to the root folder of the repo. Now it builds just fine.

**Reply by vruumllc** [reply]
Yeah, sorry. You want to change into the repo’s cc65 subdirectory in a command line, then start code from there with ‘code .’
